src/Http/SecureHttpClient.php

```diff
@@ -64,26 +60,26 @@ public function __construct($authority, KeyPair $accountKeyPair)
      */
     public function request($method, $endpoint, array $payload)
     {
-        $privateKey = $this->accountKeyPair->getPrivateKey();
+        $privateKey = $this->accountKeyPair->getPrivateKey()->getResource();
         $details = openssl_pkey_get_details($privateKey);
 
         $header = [
             'alg' => 'RS256',
             'jwk' => [
                 'kty' => 'RSA',
-                'n'   => Base64UrlSafeEncoder::encode($details['rsa']['n']),
-                'e'   => Base64UrlSafeEncoder::encode($details['rsa']['e']),
+                'n'   => Base64SafeEncoder::encode($details['rsa']['n']),
+                'e'   => Base64SafeEncoder::encode($details['rsa']['e']),
             ],
         ];
 
         $protected = $header;
         $protected['nonce'] = $this->lastResponse->getHeaderLine('Replay-Nonce');
 
-        $payload = Base64UrlSafeEncoder::encode(str_replace('\\/', '/', json_encode($payload)));
-        $protected = Base64UrlSafeEncoder::encode(json_encode($protected));
+        $payload = Base64SafeEncoder::encode(str_replace('\\/', '/', json_encode($payload)));
+        $protected = Base64SafeEncoder::encode(json_encode($protected));
 
         openssl_sign($protected.'.'.$payload, $signature, $privateKey, 'SHA256');
```

Comment:
then the extension `openssl` should not be removed from the composer's requirements